packages/react/src/utilities/ThemeProvider/makeStyles.ts

```diff
@@ -4,36 +4,72 @@ import { useWindow } from '@fluentui/react-window-provider';
 import { mergeStylesRenderer } from './styleRenderers/mergeStylesRenderer';
 import type { IStyle } from '@fluentui/style-utilities';
 import type { Theme } from '@fluentui/theme';
+import { useEffect } from 'react';
 
-const graphGet = (graphNode: Map<any, any>, path: any[]): any | undefined => {
-  for (const key of path) {
-    graphNode = graphNode.get(key);
-
-    if (!graphNode) {
-      return;
-    }
-  }
+type GraphPath = readonly [windowId: string | undefined, id: number, theme: Theme | undefined];
+export type StylesClasMapping<TStyleSet extends { [key in keyof TStyleSet]: IStyle }> = {
+  [key in keyof TStyleSet]: string;
+};
+type Graph<TStyleSet extends { [key in keyof TStyleSet]: IStyle }> = Map<
+  string | undefined,
+  Map<number, Map<Theme | undefined, { classMap: StylesClasMapping<TStyleSet>; refCount: number }>>
+>;
+
+const graphGet = <TStyleSet extends { [key in keyof TStyleSet]: IStyle }>(
+  graphNode: Graph<TStyleSet>,
+  [windowId, id, theme]: GraphPath,
+): StylesClasMapping<TStyleSet> | undefined => {
+  return graphNode.get(windowId)?.get(id)?.get(theme)?.classMap;
+};
 
-  return graphNode;
+const graphSet = <TStyleSet extends { [key in keyof TStyleSet]: IStyle }>(
+  graphNode: Graph<TStyleSet>,
+  [windowId, id, theme]: GraphPath,
+  classMap: StylesClasMapping<TStyleSet>,
+) => {
+  const windowNode =
+    graphNode.get(windowId) ??
+    new Map<number, Map<Theme | undefined, { classMap: StylesClasMapping<TStyleSet>; refCount: number }>>();
+  graphNode.set(windowId, windowNode);
+
+  const idNode =
+    windowNode.get(id) ?? new Map<Theme | undefined, { classMap: StylesClasMapping<TStyleSet>; refCount: number }>();
+  windowNode.set(id, idNode);
+
+  idNode.set(theme, { classMap, refCount: 0 });
```

Comment:
I think you would want the count to be 1 because you are not going to be calling graphRef after this

Reply:
Yes, we are, in the `useEffect` call on line 140. That way, every ref is paired with a deref, and we only purge the entry when the `refCount` is zero *after a deref*.